internal/rbd/driver/driver.go

```diff
@@ -125,17 +125,6 @@ func (r *Driver) Run(conf *util.Config) {
 			})
 	}
 
-	if k8s.RunsOnKubernetes() {
-		nodeLabels, err = k8s.GetNodeLabels(conf.NodeID)
-		if err != nil {
-			log.FatalLogMsg(err.Error())
-		}
-	}
-
-	if conf.EnableReadAffinity {
-		crushLocationMap = util.GetCrushLocationMap(conf.CrushLocationLabels, nodeLabels)
-	}
-
 	// Create GRPC servers
```

Comment:
```suggestion
	if k8s.RunsOnKubernetes() && conf.IsNodeServer {
		nodeLabels, err = k8s.GetNodeLabels(conf.NodeID)
		if err != nil {
			log.FatalLogMsg(err.Error())
		}
	}
	if conf.EnableReadAffinity {
		crushLocationMap = util.GetCrushLocationMap(conf.CrushLocationLabels, nodeLabels)
	}
	
	// Create GRPC servers
```


@nemcikjan  golangci lint is complaining that the below nest if is too complex
https://github.com/ceph/ceph-csi/actions/runs/7139935245/job/19446163983?pr=4302

Let's just add conf.NodeServer check in the same place.